### DISCUSSION TITLE
Triggers ?

### DISCUSSION THREAD

**Discussion by aurelienesteves** [bepu/bepuphysics2#256] [General]
Hello,
I don't see anything in the documentation about triggers :
I just need something as simple as the ones in Unity : an event during the Enter, an event during the exit and NO response to collisions.
How do you achieve this with BEPU ?

Thank you !

**Comment by RossNordby** [comment]
"Triggers" don't exist as a unique feature within the library; instead the lower level APIs that can be used to build that sort of thing are exposed.

One option would be to use the `INarrowPhaseCallbacks` to grab contact information and let you specify whether constraints should be created for a pair. Triggers should return false from `ConfigureContactManifold` to avoid constraint creation.

For examples of how you could use the contact data exposed in the callbacks, see the [ContactEventsDemo](https://github.com/bepu/bepuphysics2/blob/master/Demos/Demos/ContactEventsDemo.cs) or [CollisionTrackingDemo](https://github.com/bepu/bepuphysics2/blob/master/Demos/Demos/CollisionTrackingDemo.cs). The `CollisionTrackingDemo` is a bit simpler; the callbacks just grab the latest contact information and analysis is done later.

You could also use queries. Instead of having objects in the simulation collecting contacts, you can look up objects in the broad phase for a bounding box, then perform more detailed narrow phase tests using the `CollisionBatcher` on the found objects. The [CollisionQueryDemo](https://github.com/bepu/bepuphysics2/blob/master/Demos/Demos/CollisionQueryDemo.cs) shows how you could do this.

(The demos can look a little spooky at first. The `CollisionBatcher` handling in the `CollisionQueryDemo` is especially ugly. Part of this is inherent in the low-level API the demos use, but I would still like to improve some of it. Queries in particular would benefit from an unbatched codepath that would do away with most of the complexity.)
